### ISSUE TITLE
Bootstrap UI/HTML/CSS

### ISSUE THREAD

**Issue by khanshakeeb** [khanshakeeb/wad18_ae_bidding_site#3]
After receiving mockup we need to write html/css for the mockups

1. home page
2. listing page
3. detail page
4. admin panel screens
5. login/signup page/popup
6. Search pages
7. Static pages e.g about  us, contact us etc...
8. CRUD pages

Please comment on this ticket those are working on htm/css part for the project.

**Comment by fa17mcsw0005** [comment]
I will work on html/css part of project

**Comment by khanshakeeb** [comment]
@fa17mcsw0005 but need to create mockup first who is working on mockup. I believe someone else other than you

**Comment by fa17mcsw0005** [comment]
yes Faiz Shah, Usman and Bilal names you wrote on the board under mockups heading

**Comment by khanshakeeb** [comment]
HTML/CSS team please follow these mockups
https://github.com/khanshakeeb/wad18_ae_bidding_site/issues/4

Assign yourself with github issues

**Comment by khanshakeeb** [comment]
Hello frontend team how things are going with you? Any help or any place you guys stuck do message me on slack or here

**Comment by fa17mcsw0005** [comment]
ok ....sir i take task of header n footer of the page

**Comment by khanshakeeb** [comment]
@fa17mcsw0005 can you add task on github issue and assign to yourself?

**Comment by fa17mcsw0007** [comment]
Ok Sir, i take task of Home page ...

**Comment by nabiljaved** [comment]
for dynamic header and footer you can review my small working project with database 
https://mega.nz/#!WvIz2SjY!dPLt_kwPs3rM4zXokQwTgEPHhkkZjlo5cP4K7VZH9D0
Database:
https://mega.nz/#!u2QDwSAK!ysX1lkNJZIL2Nlyd8z0ZGPZ-7Fc7VRXola09vSOgImQ

**Comment by khanshakeeb** [comment]
@nabiljaved you should create a new branch and commit your work on github instead giving me external site links for code to review

**Comment by khanshakeeb** [comment]
@nabiljaved I have seen your codebase its working fine

**Comment by fa17mcsw0009** [comment]
@khanshakeeb sir i have created branch and added my work in it but when i push -u branch name or 
git push --set-upstream origin myevents it gives me this error 

remote: Permission to khanshakeeb/wad18_ae_bidding_site.git denied to fa17mcsw0009.
fatal: unable to access 'https://github.com/khanshakeeb/wad18_ae_bidding_site.git/': The requested URL returned error: 403

**Comment by fa17mcsw0009** [comment]
@khanshakeeb sir, i am working on passport js hopefully in 1 or 2 days i will have written code i just wonder why i am un able to push branch

**Comment by khanshakeeb** [comment]
> @khanshakeeb sir, i am working on passport js hopefully in 1 or 2 days i will have written code i just wonder why i am un able to push branch

Can we have a skype call/google hangout or if possible bring your laptop to univ

**Comment by fa17mcsw0009** [comment]
@khanshakeeb yes my skype is with my phone no 03082400619 will you be available tommorow sir . because today i am thinking to work on passport google strategy so when i come and see you i will have new updated work plus then u can guide me branch creation. 

sir what is your skype id: in case i am unable to be found i can add you instead

**Comment by fa17mcsw0009** [comment]
Working of Passport Authentication Google Strategy to log in with Google

1========== go to google developer create a project and enable an Api
2========= search for google+ then create credential in options web server node js
3=========select user data  and click what credential do i need???
4=========in first url paste : http://localhost:3000
5=========in second redirect url paste : http://localhost:3000/auth/google/redirect
6========create credential
7========get your client id
8=======get your client secret
9=======npm install passport-google-oauth20 in nodejs
10======import these in
const passport = require('passport');
const GoogleStrategy = require('passport-google-oauth20');
const User = require('../models/users-models');

11========(Paste this code where you have imported GoogleStrategy)
is function main client id main client id and client secret main client secret aye ga

passport.use(
 new GoogleStrategy({
     callbackURL : '/auth/google/redirect',
     clientID: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
     clientSecret: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
}, (accessToken, refreshToken, profile, done) => {
 /* call back function kuch parameters leta hai
 accesstoken : humain google say milta hai user profile ko alter karnay kay liye, read emails
 refreshToken : yeh access token ko refresh karta hai kyunkay accesstoken kuch time kay bad expire ho jata hai
 profile : yeh information lay ker ata hai
 console.log(profile.id);

 new User({
         googleId: profile.id,
         username: profile.displayName
     }).save().then((newUser) => {
         console.log('new user created: ', newUser);
     });
 })
);

12========ROUTERS
import these
const router = require('express').Router();
const passport = require('passport');

13========
router.get('/google', passport.authenticate('google', {
  /*yahan object declare kiya yahan hum nay scope propert say passport ko bta
  rahain hain keh hum user profile say kya retrieve karwana chatay hain sirf
  profile info chaiye, other email chaiye yan kiya chaiye sub bhai is array main jai ga
  */
 scope: ['profile']
 /*
 jaisay he koi /auth/google per jai ga passport usay consent screen per send ker day ga
 passprt kahay ga humain profile information chaiye kya ap allow karo gay is ap ko lenay k liye
 */
}));

14=========
REDIRECT URL
router.get('/google/redirect', passport.authenticate('google'), (req, res) => {
  res.send('you reached the callback uri');
});

15=========mongoose model and schema
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const userSchema = new Schema({
  username: String,
  googleId: String
});
const User = mongoose.model('user', userSchema);
module.exports = User;

16=========
passport strategy main you can see that i have created new User and saving user profile info with profile.id and profile.displayName
now you can save retreive and play with data we can make logic if user already signed in we dont create his profile in our database ill put more logic
